### PR TITLE
[JENKINS-36630] Convert to parent pom

### PR DESCRIPTION
**Pull request by kwhetstone**
[JENKINS-36630](https://issues.jenkins-ci.org/browse/JENKINS-36630)

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,6 +14,8 @@
     <properties>
       <findbugs-maven-plugin.version>3.0.1</findbugs-maven-plugin.version>
       <findbugs.failOnError>true</findbugs.failOnError>
+      <jenkins.version>1.609</jenkins.version>
+      <jenkins-test-harness.version>1.609</jenkins-test-harness.version>
```
Comment (jglick) [inline pom.xml]: Why not switch to the current test harness?

**Comment by recena** [comment]
@reviewbybees

**Comment by kwhetstone** [comment]
Found that there are some fundamentally incompatible pieces of 1.609 with the new htmlunit classes down in Javascript processes.  I don't think we should bump the version of core simply to work around this.  Any other opinions on this?

**Comment by andresrc** [comment]
@kwhetstone Does these problems prevent upgrading only JTH version? Can you point to a case with issues?

Some additional "pruning" in the pom may be done btw.

**Comment by kwhetstone** [comment]
@andresrc The error was attempting to save a new matrix project; certain characters that used to be valid (< and >) aren't anymore.  The problematic section disappears a few versions up.

**Comment by jglick** [comment]
> The error was attempting to save a new matrix project

What is the error? Which test fails?

> The problematic section disappears a few versions up.

Which version? Which commit fixes it?

**Comment by abayer** [comment]
@kwhetstone Do these problems only show up in the acceptance tests or the plugin compatibility tests, or should they be showing up in unit tests here as well?

**Comment by abayer** [comment]
Ah, yeah, got it on a unit test, AxisTest.submitInvalidAxisValue specifically. 
```
=== Starting submitInvalidAxisValue(hudson.matrix.AxisTest)
May 17, 2016 3:54:27 PM hudson.slaves.CommandLauncher launch
INFO: slave agent launched for slave0
May 17, 2016 3:54:32 PM com.gargoylesoftware.htmlunit.javascript.host.dom.Document createElement
INFO: createElement: Provided string '<input name="x">' contains an invalid character; '<' and '>' are not allowed
May 17, 2016 3:54:32 PM com.gargoylesoftware.htmlunit.javascript.StrictErrorReporter runtimeError
SEVERE: runtimeError: message=[String contains an invalid character] sourceName=[http://localhost:51949/jenkins/static/fc09d71e/scripts/prototype.js] line=[1927] lineSource=[null] lineOffset=[0]
May 17, 2016 3:54:37 PM hudson.matrix.MatrixProject getRootDirFor
WARNING: Cannot create directory /var/folders/pr/24nv8g910wg8vr4b4c33q34r0000gn/T/hudson8250664002364032331test/jobs/test0/configurations for the combination default
May 17, 2016 3:54:37 PM hudson.matrix.MatrixProject getRootDirFor
WARNING: Cannot create directory /var/folders/pr/24nv8g910wg8vr4b4c33q34r0000gn/T/hudson8250664002364032331test/jobs/test0/configurations for the combination default
May 17, 2016 3:54:37 PM hudson.matrix.MatrixProject getRootDirFor
WARNING: Cannot create directory /var/folders/pr/24nv8g910wg8vr4b4c33q34r0000gn/T/hudson8250664002364032331test/jobs/test0/configurations for the combination default
May 17, 2016 3:54:37 PM hudson.matrix.MatrixProject getRootDirFor
WARNING: Cannot create directory /var/folders/pr/24nv8g910wg8vr4b4c33q34r0000gn/T/hudson8250664002364032331test/jobs/test0/configurations for the combination default
May 17, 2016 3:54:38 PM com.gargoylesoftware.htmlunit.javascript.host.dom.Document createElement
INFO: createElement: Provided string '<input name="x">' contains an invalid character; '<' and '>' are not allowed
May 17, 2016 3:54:38 PM com.gargoylesoftware.htmlunit.javascript.StrictErrorReporter runtimeError
SEVERE: runtimeError: message=[String contains an invalid character] sourceName=[http://localhost:51949/jenkins/static/fc09d71e/scripts/prototype.js] line=[1927] lineSource=[null] lineOffset=[0]
May 17, 2016 3:54:39 PM org.eclipse.jetty.server.AbstractConnector doStop
INFO: Stopped ServerConnector@5ddf0d24{HTTP/1.1}{localhost:0}
May 17, 2016 3:54:39 PM org.eclipse.jetty.server.handler.ContextHandler doStop
INFO: Stopped o.e.j.w.WebAppContext@492691d7{/jenkins,file:/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test/,UNAVAILABLE}{/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test}
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 28.563 sec <<< FAILURE! - in hudson.matrix.AxisTest
```

**Comment by abayer** [comment]
That, btw, was with parent 2.9 and test harness 2.8.

**Comment by abayer** [comment]
Ignore my previous error - I think that was due to me borking something else - here's what I'm seeing now against 1.609 - the end result error still shows up against 1.651, but the exception and stacktrace along the way do not:

```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=128m; support was removed in 8.0
Running hudson.matrix.AxisTest
May 17, 2016 4:19:14 PM org.eclipse.jetty.util.log.Log initialized
INFO: Logging initialized @382ms
Exploding /Users/abayer/.m2/repository/org/jenkins-ci/main/jenkins-war/1.609/jenkins-war-1.609-war-for-test.jar into /Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test
May 17, 2016 4:19:16 PM org.eclipse.jetty.server.Server doStart
INFO: jetty-9.2.15.v20160210
May 17, 2016 4:19:16 PM org.eclipse.jetty.webapp.StandardDescriptorProcessor visitServlet
INFO: NO JSP Support for /jenkins, did not find org.eclipse.jetty.jsp.JettyJspServlet
May 17, 2016 4:19:16 PM org.eclipse.jetty.server.handler.ContextHandler doStart
INFO: Started o.e.j.w.WebAppContext@466276d8{/jenkins,file:/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test/,AVAILABLE}{/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test}
May 17, 2016 4:19:16 PM org.eclipse.jetty.server.AbstractConnector doStart
INFO: Started ServerConnector@2766ca9d{HTTP/1.1}{localhost:52827}
May 17, 2016 4:19:16 PM org.eclipse.jetty.server.Server doStart
INFO: Started @1910ms
May 17, 2016 4:19:16 PM org.jvnet.hudson.test.JenkinsRule createWebServer
INFO: Running on http://localhost:52827/jenkins/
May 17, 2016 4:19:16 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started initialization
May 17, 2016 4:19:19 PM hudson.PluginManager$1$3$1 isDuplicate
INFO: Ignoring /var/folders/pr/24nv8g910wg8vr4b4c33q34r0000gn/T/hudson1854064089099614944tmp/matrix-project.jpi because /var/folders/pr/24nv8g910wg8vr4b4c33q34r0000gn/T/hudson1854064089099614944tmp/the.jpl is already loaded
May 17, 2016 4:19:25 PM hudson.PluginManager$1$3$2$1 reactOnCycle
SEVERE: found cycle in plugin dependencies: (root=Plugin:matrix-auth, deactivating all involved) Plugin:matrix-auth -> Plugin:cloudbees-folder -> Plugin:credentials -> Plugin:matrix-auth
May 17, 2016 4:19:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Listed all plugins
May 17, 2016 4:19:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Prepared all plugins
May 17, 2016 4:19:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Started all plugins
May 17, 2016 4:19:25 PM jenkins.InitReactorRunner$1 onAttained
INFO: Augmented all extensions
May 17, 2016 4:19:26 PM jenkins.InitReactorRunner$1 onAttained
INFO: Loaded all jobs
May 17, 2016 4:19:27 PM org.jenkinsci.main.modules.sshd.SSHD start
INFO: Started SSHD at port 52830
May 17, 2016 4:19:27 PM jenkins.InitReactorRunner$1 onAttained
INFO: Completed initialization
May 17, 2016 4:19:28 PM org.eclipse.jetty.server.Server doStart
INFO: jetty-9.2.15.v20160210
May 17, 2016 4:19:28 PM org.eclipse.jetty.server.handler.ContextHandler doStart
INFO: Started o.e.j.s.ServletContextHandler@50eca7c6{/,null,AVAILABLE}
May 17, 2016 4:19:28 PM org.eclipse.jetty.server.AbstractConnector doStart
INFO: Started ServerConnector@28d6290{HTTP/1.1}{0.0.0.0:52832}
May 17, 2016 4:19:28 PM org.eclipse.jetty.server.Server doStart
INFO: Started @13696ms
=== Starting submitInvalidAxisValue(hudson.matrix.AxisTest)
May 17, 2016 4:19:30 PM hudson.slaves.CommandLauncher launch
INFO: slave agent launched for slave0
May 17, 2016 4:19:35 PM com.gargoylesoftware.htmlunit.javascript.host.dom.Document createElement
INFO: createElement: Provided string '<input name="x">' contains an invalid character; '<' and '>' are not allowed
May 17, 2016 4:19:35 PM com.gargoylesoftware.htmlunit.javascript.StrictErrorReporter runtimeError
SEVERE: runtimeError: message=[String contains an invalid character] sourceName=[http://localhost:52827/jenkins/static/ff90da60/scripts/prototype.js] line=[1927] lineSource=[null] lineOffset=[0]
May 17, 2016 4:19:39 PM com.gargoylesoftware.htmlunit.javascript.background.JavaScriptJobManagerImpl runSingleJob
SEVERE: Job run failed with unexpected RuntimeException: TypeError: Cannot find function on in object [object HTMLSpanElement]. (http://localhost:52827/jenkins/static/ff90da60/scripts/hudson-behavior.js#2806)
======= EXCEPTION START ========
Exception class=[net.sourceforge.htmlunit.corejs.javascript.JavaScriptException]
com.gargoylesoftware.htmlunit.ScriptException: TypeError: Cannot find function on in object [object HTMLSpanElement]. (http://localhost:52827/jenkins/static/ff90da60/scripts/hudson-behavior.js#2806)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$HtmlUnitContextAction.run(JavaScriptEngine.java:865)
	at net.sourceforge.htmlunit.corejs.javascript.Context.call(Context.java:628)
	at net.sourceforge.htmlunit.corejs.javascript.ContextFactory.call(ContextFactory.java:513)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine.callFunction(JavaScriptEngine.java:796)
	at com.gargoylesoftware.htmlunit.javascript.host.xml.XMLHttpRequest.setState(XMLHttpRequest.java:224)
	at com.gargoylesoftware.htmlunit.javascript.host.xml.XMLHttpRequest.doSend(XMLHttpRequest.java:775)
	at com.gargoylesoftware.htmlunit.javascript.host.xml.XMLHttpRequest.access$000(XMLHttpRequest.java:105)
	at com.gargoylesoftware.htmlunit.javascript.host.xml.XMLHttpRequest$1.run(XMLHttpRequest.java:630)
	at net.sourceforge.htmlunit.corejs.javascript.Context.call(Context.java:628)
	at net.sourceforge.htmlunit.corejs.javascript.ContextFactory.call(ContextFactory.java:513)
	at com.gargoylesoftware.htmlunit.javascript.background.JavascriptXMLHttpRequestJob.run(JavascriptXMLHttpRequestJob.java:37)
	at com.gargoylesoftware.htmlunit.javascript.background.JavaScriptJobManagerImpl.runSingleJob(JavaScriptJobManagerImpl.java:428)
	at com.gargoylesoftware.htmlunit.javascript.background.DefaultJavaScriptExecutor.run(DefaultJavaScriptExecutor.java:168)
	at java.lang.Thread.run(Thread.java:744)
Caused by: net.sourceforge.htmlunit.corejs.javascript.JavaScriptException: TypeError: Cannot find function on in object [object HTMLSpanElement]. (http://localhost:52827/jenkins/static/ff90da60/scripts/hudson-behavior.js#2806)
	at net.sourceforge.htmlunit.corejs.javascript.Interpreter.interpretLoop(Interpreter.java:1006)
	at net.sourceforge.htmlunit.corejs.javascript.Interpreter.interpret(Interpreter.java:798)
	at net.sourceforge.htmlunit.corejs.javascript.InterpretedFunction.call(InterpretedFunction.java:105)
	at net.sourceforge.htmlunit.corejs.javascript.ContextFactory.doTopCall(ContextFactory.java:411)
	at com.gargoylesoftware.htmlunit.javascript.HtmlUnitContextFactory.doTopCall(HtmlUnitContextFactory.java:310)
	at net.sourceforge.htmlunit.corejs.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3286)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$4.doRun(JavaScriptEngine.java:789)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$HtmlUnitContextAction.run(JavaScriptEngine.java:850)
	... 13 more
JavaScriptException value = TypeError: Cannot find function on in object [object HTMLSpanElement].
== CALLING JAVASCRIPT ==
  function () {
      var a = merge(args, arguments);
      return __method.apply(context, a);
  }
======= EXCEPTION END ========

May 17, 2016 4:19:39 PM org.eclipse.jetty.server.AbstractConnector doStop
INFO: Stopped ServerConnector@2766ca9d{HTTP/1.1}{localhost:0}
May 17, 2016 4:19:39 PM org.eclipse.jetty.server.handler.ContextHandler doStop
INFO: Stopped o.e.j.w.WebAppContext@466276d8{/jenkins,file:/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test/,UNAVAILABLE}{/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test}
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 25.326 sec <<< FAILURE! - in hudson.matrix.AxisTest
submitInvalidAxisValue(hudson.matrix.AxisTest)  Time elapsed: 24.801 sec  <<< ERROR!
com.gargoylesoftware.htmlunit.ElementNotFoundException: elementName=[input] attributeName=[name] attributeValue=[_.name]
	at hudson.matrix.AxisTest.submitInvalidAxisValue(AxisTest.java:90)
```

**Comment by abayer** [comment]
Latest and greatest failure - all kinds of weird environmental noise, but this one seems to be consistent:

```
=== Starting submitInvalidAxisValue(hudson.matrix.AxisTest)
May 17, 2016 4:24:45 PM hudson.slaves.CommandLauncher launch
INFO: slave agent launched for slave0
May 17, 2016 4:24:52 PM com.gargoylesoftware.htmlunit.javascript.host.dom.Document createElement
INFO: createElement: Provided string '<input name="x">' contains an invalid character; '<' and '>' are not allowed
May 17, 2016 4:24:52 PM com.gargoylesoftware.htmlunit.javascript.StrictErrorReporter runtimeError
SEVERE: runtimeError: message=[String contains an invalid character] sourceName=[http://localhost:53033/jenkins/static/71411275/scripts/prototype.js] line=[1927] lineSource=[null] lineOffset=[0]
May 17, 2016 4:24:57 PM org.eclipse.jetty.server.AbstractConnector doStop
INFO: Stopped ServerConnector@6dab9b6d{HTTP/1.1}{localhost:0}
May 17, 2016 4:24:57 PM org.eclipse.jetty.server.handler.ContextHandler doStop
INFO: Stopped o.e.j.w.WebAppContext@49964d75{/jenkins,file:/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test/,UNAVAILABLE}{/Users/abayer/IdeaProjects/matrix-project-plugin/target/jenkins-for-test}
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
WARN: The method class org.apache.commons.logging.impl.SLF4JLogFactory#release() was invoked.
WARN: Please see http://www.slf4j.org/codes.html#release for an explanation.
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 30.671 sec <<< FAILURE! - in hudson.matrix.AxisTest
submitInvalidAxisValue(hudson.matrix.AxisTest)  Time elapsed: 30.105 sec  <<< ERROR!
com.gargoylesoftware.htmlunit.ElementNotFoundException: elementName=[input] attributeName=[name] attributeValue=[_.name]
	at hudson.matrix.AxisTest.submitInvalidAxisValue(AxisTest.java:90)
```

**Comment by abayer** [comment]
And I think the invalid character thing is a red herring - that vanishes when I switch to 1.651, but the end error is still the same.

**Comment by amuniz** [comment]
With the current configuration I only get on test fail, [which is marked as `@RandomlyFails`](https://github.com/jenkinsci/matrix-project-plugin/blob/510aa05c7909e421f2ea79465280948b613dece3/src/test/java/hudson/matrix/MatrixProjectTest.java#L136).

**Comment by jglick** [comment]
`@RandomlyFails` should be considered deprecated. The parent POM sets up Surefire to rerun flaky tests, which is better—genuine regressions get caught.

**Comment by jglick** [comment]
In other words, if the test _consistently_ fails, figure out if it is a real problem, and either fix it or delete the test. If the test has been _observed_ to fail on occasion, but you cannot reproduce it locally, just delete `@RandomlyFails` and let it flake once in a while and get rerun.

**Comment by kwhetstone** [comment]
Found that the problem here is still a Javascript error, just not the one originally presented.  During matrix creation, the item has an issue creating the matrix `TypeError` on the `HTMLSpanElement` while running `runSingleJob`.  The error makes sense, just how it's gotten to this point doesn't to me.

At least if this has been converted to the parent pom, it'll be easier to test against the various versions of Jenkins.

**Comment by recena** [comment]
@kwhetstone Is there any OSS JIRA issue for that?

File: pom.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>1.609</version>
+        <version>2.3</version>
```
Comment (recena) [inline pom.xml]: Maybe `2.12`?

**Comment by kwhetstone** [comment]
Still have this issue after trying 2.12.  @jglick @recena @andresrc @amuniz at this point, what do we want to do here?

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: 2.16

File: pom.xml (hunk 1 above)
Comment (oleg-nenashev) [inline pom.xml]: +1

**Comment by jglick** [comment]
> how it's gotten to this point doesn't to me

Debug…

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: 2.17 and counting

**Comment by recena** [comment]
@kwhetstone ping?

**Comment by jglick** [comment]
This PR is broken, submit a fresh one.

**Pull request closed without merging**